### ISSUE TITLE
0xc0000142 error on powershell (running "cmd.exe /b /c start /b /min powershell.exe" inside WSL)

### ISSUE THREAD

**Issue by emdns** [microsoft/WSL#9542]
### Version

10.0.22000.1455

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.79.1

### Distro Version

Debian and Ubuntu

### Other Software

Powershell 5.1.22000.1335

### Repro Steps

Launch wsl
Inside wsl bash shell type the command:
`cmd.exe /b /c start /b /min powershell.exe`

### Expected Behavior

A powershell running

### Actual Behavior

Powershell either silently fails to run or fails to run with the following popup:

![image](https://user-images.githubusercontent.com/123670936/214932819-0b721277-558e-4ff7-bae0-7674a0933c79.png)

### Diagnostic Logs

[WslLogs-2023-01-26_11-27-25.zip](https://github.com/microsoft/WSL/files/10512780/WslLogs-2023-01-26_11-27-25.zip)

**Comment by emdns** [comment]
@DHowett
(for vis)

**Comment by DHowett** [comment]
@emdns Thanks for the tag. I think that this was fixed by a recent change we've made in Windows and will need some time to propagate out to all of our API consumers.

The specific issue:

`cmd.exe` is launched in a console session. `cmd.exe /c start /b xyz` launches application `xyz` inside the same console session _and then immediately exits._

This means that depending on how quickly `xyz` starts up, _it might not have a console any more._ Because the session leader, cmd, just exited... which took down the console. :)
